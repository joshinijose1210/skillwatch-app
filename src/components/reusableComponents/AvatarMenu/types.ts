export interface AvatarMenuProps {
    open: boolean;
    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
}
