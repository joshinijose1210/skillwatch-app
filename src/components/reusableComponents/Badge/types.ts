export type Props = {
    children?: React.ReactNode;
    bgColor?: string;
    pd?: string;
    borderRadius?: string;
};
