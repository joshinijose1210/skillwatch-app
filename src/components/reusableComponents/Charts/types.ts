import { ReactNode } from 'react';

export interface LineChartProps {
    rating: number;
    ratingBy?: string;
}

export interface BarChartProps {
    rating: number;
}

export type DataChunk = {
    id: string;
    label: string;
    value: number;
    color: string;
};

export type PieChartPropTypes = {
    data: DataChunk[];
    title: ReactNode;
    noData?: boolean;
    padAngle?: number;
    enableArcLabels?: boolean;
    enableArcLinkLabels?: boolean;
    margin?: number;
    isInteractive?: boolean;
};
