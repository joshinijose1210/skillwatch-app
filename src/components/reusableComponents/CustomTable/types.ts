import { RowHoverActionsType, TableColumnConfig } from '@medly-components/core';
import { SortOrder } from '@medly-components/core/dist/es/components/Table/types';

interface CustomTableBaseProps {
    children?: React.ReactNode;
    columns: TableColumnConfig[];
    count: number;
    data: any;
    isLoading: boolean;
    activePage: number;
    rowAction?: RowHoverActionsType;
    defaultSortField?: string;
    defaultSortOrder?: SortOrder;
    tableKey?: string;
    itemsPerPage?: number;
    setSortOrder?: (order: string) => void;
    withMinimap?: boolean;
}

interface WithPaginationProps extends CustomTableBaseProps {
    withPagination: true;
    handlePageChange: (page: number) => void;
}

interface WithoutPaginationProps extends CustomTableBaseProps {
    withPagination?: false;
    handlePageChange?: (page: number) => void;
}

// Union type
export type CustomTableProps = WithPaginationProps | WithoutPaginationProps;
