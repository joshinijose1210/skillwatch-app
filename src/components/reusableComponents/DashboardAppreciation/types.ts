export interface Appreciation {
    isExternalFeedback?: boolean;
    externalFeedbackFromEmailId?: string;
    appreciationToEmployeeId: string;
    appreciationToFirstName: string;
    appreciationFromId: string;
    appreciationToId: string;
    appreciationToLastName: string;
    appreciation: string;
    appreciationFromEmployeeId: string;
    appreciationFromRoleName: string;
    appreciationFromFirstName: string;
    appreciationFromLastName: string;
    appreciationToRoleName: string;
    submitDate: Date;
    isDummy?: boolean;
}

export type DashboardAppreciationPropType = {
    reviewCycleId: number;
};
