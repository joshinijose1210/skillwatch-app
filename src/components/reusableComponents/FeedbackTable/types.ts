import { TableColumnConfig } from '@medly-components/core';
import { Option } from '@medly-components/core/dist/es/components/SearchBox/types';
import { TableState } from '@medly-components/core/dist/es/components/Table/types';

export interface FeedbackTableProps {
    action: string;
    data?: any;
    columns: TableColumnConfig[];
    defaultSortOrder?: 'asc' | 'desc';
    defaultSortField?: string;
    onSort?: ({ activePage, sortField, sortOrder }: TableState) => void;
    options: Option[];
    onNameSelection: (value: string[] | any[]) => void;
    values?: string[];
    isLoading: boolean;
}
