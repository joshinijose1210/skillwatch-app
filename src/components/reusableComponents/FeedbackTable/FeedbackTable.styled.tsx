import styled from 'styled-components';

export const StyledTitle = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 2rem;
`;
