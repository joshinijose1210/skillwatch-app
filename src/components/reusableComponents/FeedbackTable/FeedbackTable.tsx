import { StyledMultiSelect, StyledTable } from '@common';
import { PageContent } from '@components/layout';
import { Text } from '@medly-components/core';
import { filterData, SortOrder } from '@utils/filterData';
import { FC, useCallback, useState } from 'react';
import { StyledTitle } from './FeedbackTable.styled';
import { FeedbackTableProps } from './types';
import { SortHandlerProps } from '@common/types';

const FeedbackTable: FC<FeedbackTableProps> = (props: FeedbackTableProps) => {
    const { action, data, columns, options, onNameSelection, defaultSortOrder, values, defaultSortField, isLoading } = props;
    const [order, setOrder] = useState<SortOrder>('desc'),
        [sortingField, setSortingField] = useState<string>('sr_no');

    const sortedData = filterData(sortingField, order, data);

    const handleSortData = useCallback(({ sortField, sortOrder }: SortHandlerProps) => {
        if (sortField && sortOrder) {
            setSortingField(sortField);
            setOrder(sortOrder);
        }
    }, []);
    return (
        <PageContent>
            <StyledTitle data-testid="title">
                <Text textVariant="h2" textWeight="Medium">
                    {action} Feedbacks
                </Text>
            </StyledTitle>
            <StyledTitle>
                <StyledMultiSelect
                    label="Filter By Name"
                    options={options}
                    values={values}
                    onChange={value => onNameSelection(value)}
                    variant={'outlined'}
                    size="M"
                    minWidth="30rem"
                />
            </StyledTitle>

            <StyledTable
                defaultSortOrder={defaultSortOrder}
                defaultSortField={defaultSortField}
                withPagination={true}
                totalItems={data.length}
                itemsPerPage={10}
                defaultActivePage={1}
                data={sortedData}
                onSort={handleSortData}
                columns={columns}
                isLoading={isLoading}
            />
        </PageContent>
    );
};

export default FeedbackTable;
