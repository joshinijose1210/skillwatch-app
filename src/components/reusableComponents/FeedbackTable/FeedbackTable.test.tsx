import { PageContent } from '@components/layout';
import { RTKRender } from '@utils/test-utils';
import FeedbackTable from './FeedbackTable';
import { FeedbackTableProps } from './types';

const defaultProps: FeedbackTableProps = {
    columns: [{ title: 'dummyTitle', field: 'dummyField', sortable: true }],
    data: [],
    action: 'dummyAction',
    onSort: jest.fn(),
    options: [{ label: 'dummy', value: 'dummy', selected: true }],
    values: ['dummy1', 'dummy2'],
    onNameSelection: jest.fn(),
    isLoading: false
};
const renderer = (props = defaultProps) =>
    RTKRender(
        <PageContent>
            <FeedbackTable {...props} />
        </PageContent>
    );

describe('FeedbackTable', () => {
    it('should render properly with default props', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });
});
