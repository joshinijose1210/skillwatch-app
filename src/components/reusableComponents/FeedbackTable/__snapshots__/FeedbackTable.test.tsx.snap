// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackTable should render properly with default props 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <main
      class="sc-1ehavxz-0 dDyhBJ"
    >
      <div
        class="sc-v6r37y-0 cqkEzK"
        data-testid="title"
      >
        <h2
          class="sc-1ntgl3l-0 hJdvRC"
          textalign="initial"
          textcolor="inherit"
          textvariant="h2"
          textweight="Medium"
        >
          dummyAction
           Feedbacks
        </h2>
      </div>
      <div
        class="sc-v6r37y-0 cqkEzK"
      >
        <div
          class="sc-11tbpdo-0 iUudLF sc-tkiih6-14 gSveWN"
          id="filterby name-wrapper"
          minwidth="30rem"
          variant="outlined"
        >
          <div
            class="sc-17d4s39-0 ihsYRx"
            id="filterby name-input-wrapper"
            minwidth="30rem"
          >
            <div
              class="sc-19px6zp-0 BrSGZ"
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="filterby name-helper-text"
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  errortext=""
                  id="filterby name-input"
                  inputsize="M"
                  placeholder="2 options selected"
                  type="text"
                  value=""
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="filterby name-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Filter By Name
                </label>
              </div>
              <div
                class="sc-4vbg19-0 hFRmRP"
                id="filterby name-count"
              >
                <svg
                  class="sc-1pf3x8a-0 klDMsE"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                    fill="#000"
                  />
                </svg>
              </div>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="sc-1msmxxr-0 iWEzjW"
      >
        dummyTitle
      </div>
      <table
        actions=""
        class="sc-1msmxxr-1 jwFsPF sc-tkiih6-2 guTFzC"
        columns="[object Object]"
        defaultactivepage="1"
        defaultsortorder="desc"
        itemsperpage="10"
        rowclickdisablekey=""
        tabindex="-1"
        totalitems="0"
      >
        <thead
          class="sc-uxzsxp-0 haNltq"
        >
          <tr
            class="sc-1jr91bg-0 hJzMcO"
            gridtemplatecolumns=" minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr);"
          >
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 cZgEuC"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  dummyTitle
                </span>
                <svg
                  class="sc-1pf3x8a-0 bMQNVa"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.229 6.935a1 1 0 0 1 1.543 0l2.208 2.679a1 1 0 0 1-.77 1.636H9.79a1 1 0 0 1-.77-1.636l2.209-2.679zM12.771 17.065a1 1 0 0 1-1.543 0L9.02 14.386a1 1 0 0 1 .772-1.636h4.418a1 1 0 0 1 .771 1.636l-2.209 2.679z"
                    fill="#435465"
                  />
                </svg>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
          </tr>
        </thead>
        <tbody
          class="sc-c7cniy-0 kShdfA"
        >
          <tr
            class="sc-1yxujaa-0 sc-1yxujaa-2 cfQtmZ gtGPdU"
            gridtemplatecolumns=" minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr);"
          >
            <td
              class="sc-1yxujaa-1 hmPqUc"
              tablesize="M"
              width="0"
            >
              No result
            </td>
          </tr>
        </tbody>
        <tfoot
          class="sc-1l1oiao-1 bwlyAP"
          tablesize="M"
        >
          <tr
            class="sc-1l1oiao-2 kUNtAH"
          >
            <td>
              <ul
                class="sc-8e7n0t-0 fAnmfk sc-1dmgj8o-0 fgZbRq sc-1l1oiao-0 dExqdq"
                data-testid="foot-pagination"
                spacings="[object Object]"
                tablesize="M"
                variant="horizontal"
              >
                <li>
                  <button
                    class="sc-1dmgj8o-1 sc-1dmgj8o-3 bmhrYt bplbKR"
                    disabled=""
                    type="button"
                  >
                    <svg
                      class="sc-1pf3x8a-0 klDMsE"
                      fill="none"
                      height="1em"
                      title="prev"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15 6.705a.996.996 0 0 0-1.41 0L9 11.295a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41l-3.88-3.89L15 8.115c.39-.39.38-1.03 0-1.41z"
                        fill="#000"
                        fill-opacity=".54"
                      />
                    </svg>
                  </button>
                </li>
                <li>
                  <button
                    class="sc-1dmgj8o-1 sc-1dmgj8o-3 bmhrYt bplbKR"
                    disabled=""
                    type="button"
                  >
                    <svg
                      class="sc-1pf3x8a-0 klDMsE"
                      fill="none"
                      height="1em"
                      title="next"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 6.71a.996.996 0 0 0 0 1.41L12.88 12 9 15.88a.996.996 0 1 0 1.41 1.41L15 12.7a.996.996 0 0 0 0-1.41L10.41 6.7c-.38-.38-1.02-.38-1.41.01z"
                        fill="#000"
                        fill-opacity=".54"
                      />
                    </svg>
                  </button>
                </li>
              </ul>
            </td>
            <td>
              <span
                class="sc-1ntgl3l-0 hGagSU"
                textalign="initial"
                textcolor="inherit"
              >
                Viewing
              </span>
              <strong
                class="sc-1ntgl3l-0 dDvaST"
                textalign="initial"
                textcolor="inherit"
                textweight="Strong"
              >
                 0 - 0 
              </strong>
              <span
                class="sc-1ntgl3l-0 hGagSU"
                textalign="initial"
                textcolor="inherit"
              >
                of
              </span>
              <strong
                class="sc-1ntgl3l-0 dDvaST"
                textalign="initial"
                textcolor="inherit"
                textweight="Strong"
              >
                 0 
              </strong>
              <span
                class="sc-1ntgl3l-0 hGagSU"
                textalign="initial"
                textcolor="inherit"
              >
                Results
              </span>
            </td>
          </tr>
        </tfoot>
      </table>
    </main>
  </main>
</div>
`;
